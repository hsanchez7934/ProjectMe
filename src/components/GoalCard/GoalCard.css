@import url(https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700);
.goal-card {
  background: linear-gradient(0deg, #c0b283, 35%, #000);
  border: 1px solid #000;
  border-radius: 15px;
  display: flex;
  flex-direction: row;
  height: 300px;
  margin-bottom: 40px;
  width: 650px; }
  .goal-card .card-left-side {
    height: 100%;
    position: relative;
    width: 60%; }
    .goal-card .card-left-side .goal-title-container {
      align-items: center;
      background: linear-gradient(0deg, #373737, 35%, #c0b283);
      box-shadow: 7px 7px 7px 1px #373737;
      display: flex;
      height: 60%;
      justify-content: center;
      left: -5px;
      overflow: scroll;
      position: absolute;
      top: -15px;
      width: 100%; }
      .goal-card .card-left-side .goal-title-container .goal-title {
        background-color: #000;
        color: #f4f4f4;
        font-size: 26px;
        height: 90%;
        margin: 0 auto;
        overflow: scroll;
        padding: 60px 10px;
        text-align: center;
        width: 95%; }
    .goal-card .card-left-side .goal-card-date-container {
      color: #373737;
      height: 40%;
      margin-top: 150px;
      padding: 30px 25px 0px 25px; }
      .goal-card .card-left-side .goal-card-date-container .weekday {
        color: #f4f4f4;
        font-size: 27px;
        margin: 0;
        padding-bottom: 10px; }
      .goal-card .card-left-side .goal-card-date-container .date {
        font-size: 15px;
        margin: 0;
        padding-bottom: 2px; }
      .goal-card .card-left-side .goal-card-date-container .time {
        font-size: 12px;
        margin: 0; }
  .goal-card .card-right-side {
    height: 100%;
    width: 40%; }
    .goal-card .card-right-side .goal-description {
      border-left: 1px solid #f4f4f4;
      border-bottom: 1px solid #f4f4f4;
      color: #f4f4f4;
      font-weight: lighter;
      height: 65%;
      margin: 15px auto;
      overflow: scroll;
      padding: 15px 10px;
      text-align: center;
      width: 80%; }
    .goal-card .card-right-side .delete-button {
      background-color: #000;
      border-width: 0px;
      color: #f4f4f4;
      font-size: 15px;
      display: block;
      height: 40px;
      margin: 0 auto;
      width: 100px; }
    .goal-card .card-right-side .delete-button:hover {
      background-color: #c0b283;
      border: 4px solid #000;
      cursor: pointer;
      color: #000;
      font-weight: bold; }
