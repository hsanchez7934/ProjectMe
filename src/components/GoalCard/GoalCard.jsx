import React from 'react';
import { connect } from 'react-redux';

const GoalCard = ({ title, body }) => (
  <article className='goal-card'>
    <h3 className='goal-title'> {title} </h3>
    <p className='goal-description'> {body} </p>
  </article>
)

export default GoalCard;
