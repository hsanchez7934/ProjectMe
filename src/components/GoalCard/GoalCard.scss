@import "../../styles/opensans.scss";
@import "../../styles/colors.scss";

.goal-card {
  background: linear-gradient(0deg, $palegold, 35%, $black);
  border: 1px solid $black;
  border-radius: 15px;
  display: flex;
  flex-direction: row;
  height: 300px;
  margin-bottom: 40px;
  width: 650px;
  .card-left-side {
    height: 100%;
    position: relative;
    width: 60%;
    .goal-title-container {
      align-items: center;
      background: linear-gradient(0deg, $charcoal, 35%, $palegold);
      box-shadow: 7px 7px 7px 1px $charcoal;
      display: flex;
      height: 60%;
      justify-content: center;
      left: -5px;
      overflow: scroll;
      position: absolute;
      top: -15px;
      width: 100%;
      .goal-title {
        background-color: $black;
        color: $paper;
        font-size: 26px;
        height: 90%;
        margin: 0 auto;
        overflow: scroll;
        padding: 60px 10px;
        text-align: center;
        width: 95%;
      }
    }
    .goal-card-date-container {
      color: $charcoal;
      height: 40%;
      margin-top: 150px;
      padding: 30px 25px 0px 25px;
      .weekday {
        color: $paper;
        font-size: 27px;
        margin: 0;
        padding-bottom: 10px;
      }
      .date {
        font-size: 15px;
        margin: 0;
        padding-bottom: 2px;
      }
      .time {
        font-size: 12px;
        margin: 0;
      }
    }
  }
  .card-right-side {
    height: 100%;
    width: 40%;
    .goal-description {
      border-left: 1px solid $paper;
      border-bottom: 1px solid $paper;
      color: $paper;
      font-weight: lighter;
      height: 65%;
      margin: 15px auto;
      overflow: scroll;
      padding: 15px 10px;
      text-align: center;
      width: 80%;
    }
    .delete-button {
      background-color: $black;
      border-width: 0px;
      color: $paper;
      font-size: 15px;
      display: block;
      height: 40px;
      margin: 0 auto;
      width: 100px;
    }
    .delete-button:hover {
      background-color: $palegold;
      border: 4px solid $black;
      cursor: pointer;
      color: $black;
      font-weight: bold;
    }
  }
}
