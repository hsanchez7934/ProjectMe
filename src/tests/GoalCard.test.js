import React from 'react';
import { mount } from 'enzyme';
import 'jest';
import config from './testSetup.js';
import GoalCard from '../components/GoalCard/GoalCard.jsx';

describe(`GoalCard unit testing`, () => {
  let wrapper;
  let goal;
  let mockFunction;

  beforeEach(() => {

    goal = {
      title: 'Workout',
      body: 'Go to gym',
      id: '1234',
      date: 'November 6, 2017',
      time: '07:27:57'
    };

    mockFunction = jest.fn();

    wrapper = mount(
      <GoalCard
        goal={goal}
        removeGoal={mockFunction} />
    );
  });

  test(`should create an instance of GoalCard`, () => {
    expect(wrapper.exists()).toEqual(true);
  });

  test(`should match snapshot`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
