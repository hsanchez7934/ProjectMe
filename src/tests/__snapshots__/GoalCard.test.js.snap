// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalCard unit testing should match snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <GoalCard
    goal={
      Object {
        "body": "Go to gym",
        "date": "November 6, 2017",
        "id": "1234",
        "time": "07:27:57",
        "title": "Workout",
      }
    }
    removeGoal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "goal": Object {
        "body": "Go to gym",
        "date": "November 6, 2017",
        "id": "1234",
        "time": "07:27:57",
        "title": "Workout",
      },
      "removeGoal": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <article
        class="goal-card"
        id="1234"
      >
        <section
          class="card-left-side"
        >
          <section
            class="goal-title-container"
          >
            <h3
              class="goal-title"
            >
              Workout
            </h3>
          </section>
          <section
            class="goal-card-date-container"
          >
            <p
              class="weekday"
            />
            <p
              class="date"
            >
              November 6, 2017
            </p>
            <p
              class="time"
            >
              07:27:57
            </p>
          </section>
        </section>
        <section
          class="card-right-side"
        >
          <p
            class="goal-description"
          >
            Go to gym
          </p>
          <button
            class="delete-button"
          >
            Delete
          </button>
        </section>
      </article>,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="card-left-side"
          >
            <section
              className="goal-title-container"
            >
              <h3
                className="goal-title"
              >
                Workout
              </h3>
            </section>
            <section
              className="goal-card-date-container"
            >
              <p
                className="weekday"
              />
              <p
                className="date"
              >
                November 6, 2017
              </p>
              <p
                className="time"
              >
                07:27:57
              </p>
            </section>
          </section>,
          <section
            className="card-right-side"
          >
            <p
              className="goal-description"
            >
              Go to gym
            </p>
            <button
              className="delete-button"
              onClick={[Function]}
            >
              Delete
            </button>
          </section>,
        ],
        "className": "goal-card",
        "id": "1234",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <section
            class="card-left-side"
          >
            <section
              class="goal-title-container"
            >
              <h3
                class="goal-title"
              >
                Workout
              </h3>
            </section>
            <section
              class="goal-card-date-container"
            >
              <p
                class="weekday"
              />
              <p
                class="date"
              >
                November 6, 2017
              </p>
              <p
                class="time"
              >
                07:27:57
              </p>
            </section>
          </section>,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <section
                className="goal-title-container"
              >
                <h3
                  className="goal-title"
                >
                  Workout
                </h3>
              </section>,
              <section
                className="goal-card-date-container"
              >
                <p
                  className="weekday"
                />
                <p
                  className="date"
                >
                  November 6, 2017
                </p>
                <p
                  className="time"
                >
                  07:27:57
                </p>
              </section>,
            ],
            "className": "card-left-side",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <section
                class="goal-title-container"
              >
                <h3
                  class="goal-title"
                >
                  Workout
                </h3>
              </section>,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <h3
                  className="goal-title"
                >
                  Workout
                </h3>,
                "className": "goal-title-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <h3
                    class="goal-title"
                  >
                    Workout
                  </h3>,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Workout",
                    "className": "goal-title",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Workout",
                  ],
                  "type": "h3",
                },
              ],
              "type": "section",
            },
            Object {
              "instance": <section
                class="goal-card-date-container"
              >
                <p
                  class="weekday"
                />
                <p
                  class="date"
                >
                  November 6, 2017
                </p>
                <p
                  class="time"
                >
                  07:27:57
                </p>
              </section>,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="weekday"
                  />,
                  <p
                    className="date"
                  >
                    November 6, 2017
                  </p>,
                  <p
                    className="time"
                  >
                    07:27:57
                  </p>,
                ],
                "className": "goal-card-date-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <p
                    class="weekday"
                  />,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "weekday",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <p
                    class="date"
                  >
                    November 6, 2017
                  </p>,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "November 6, 2017",
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": Array [
                    "November 6, 2017",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <p
                    class="time"
                  >
                    07:27:57
                  </p>,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "07:27:57",
                    "className": "time",
                  },
                  "ref": null,
                  "rendered": Array [
                    "07:27:57",
                  ],
                  "type": "p",
                },
              ],
              "type": "section",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": <section
            class="card-right-side"
          >
            <p
              class="goal-description"
            >
              Go to gym
            </p>
            <button
              class="delete-button"
            >
              Delete
            </button>
          </section>,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="goal-description"
              >
                Go to gym
              </p>,
              <button
                className="delete-button"
                onClick={[Function]}
              >
                Delete
              </button>,
            ],
            "className": "card-right-side",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <p
                class="goal-description"
              >
                Go to gym
              </p>,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Go to gym",
                "className": "goal-description",
              },
              "ref": null,
              "rendered": Array [
                "Go to gym",
              ],
              "type": "p",
            },
            Object {
              "instance": <button
                class="delete-button"
              >
                Delete
              </button>,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "className": "delete-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Delete",
              ],
              "type": "button",
            },
          ],
          "type": "section",
        },
      ],
      "type": "article",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "goal": Object {
          "body": "Go to gym",
          "date": "November 6, 2017",
          "id": "1234",
          "time": "07:27:57",
          "title": "Workout",
        },
        "removeGoal": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <article
          class="goal-card"
          id="1234"
        >
          <section
            class="card-left-side"
          >
            <section
              class="goal-title-container"
            >
              <h3
                class="goal-title"
              >
                Workout
              </h3>
            </section>
            <section
              class="goal-card-date-container"
            >
              <p
                class="weekday"
              />
              <p
                class="date"
              >
                November 6, 2017
              </p>
              <p
                class="time"
              >
                07:27:57
              </p>
            </section>
          </section>
          <section
            class="card-right-side"
          >
            <p
              class="goal-description"
            >
              Go to gym
            </p>
            <button
              class="delete-button"
            >
              Delete
            </button>
          </section>
        </article>,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="card-left-side"
            >
              <section
                className="goal-title-container"
              >
                <h3
                  className="goal-title"
                >
                  Workout
                </h3>
              </section>
              <section
                className="goal-card-date-container"
              >
                <p
                  className="weekday"
                />
                <p
                  className="date"
                >
                  November 6, 2017
                </p>
                <p
                  className="time"
                >
                  07:27:57
                </p>
              </section>
            </section>,
            <section
              className="card-right-side"
            >
              <p
                className="goal-description"
              >
                Go to gym
              </p>
              <button
                className="delete-button"
                onClick={[Function]}
              >
                Delete
              </button>
            </section>,
          ],
          "className": "goal-card",
          "id": "1234",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <section
              class="card-left-side"
            >
              <section
                class="goal-title-container"
              >
                <h3
                  class="goal-title"
                >
                  Workout
                </h3>
              </section>
              <section
                class="goal-card-date-container"
              >
                <p
                  class="weekday"
                />
                <p
                  class="date"
                >
                  November 6, 2017
                </p>
                <p
                  class="time"
                >
                  07:27:57
                </p>
              </section>
            </section>,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <section
                  className="goal-title-container"
                >
                  <h3
                    className="goal-title"
                  >
                    Workout
                  </h3>
                </section>,
                <section
                  className="goal-card-date-container"
                >
                  <p
                    className="weekday"
                  />
                  <p
                    className="date"
                  >
                    November 6, 2017
                  </p>
                  <p
                    className="time"
                  >
                    07:27:57
                  </p>
                </section>,
              ],
              "className": "card-left-side",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <section
                  class="goal-title-container"
                >
                  <h3
                    class="goal-title"
                  >
                    Workout
                  </h3>
                </section>,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <h3
                    className="goal-title"
                  >
                    Workout
                  </h3>,
                  "className": "goal-title-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <h3
                      class="goal-title"
                    >
                      Workout
                    </h3>,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Workout",
                      "className": "goal-title",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Workout",
                    ],
                    "type": "h3",
                  },
                ],
                "type": "section",
              },
              Object {
                "instance": <section
                  class="goal-card-date-container"
                >
                  <p
                    class="weekday"
                  />
                  <p
                    class="date"
                  >
                    November 6, 2017
                  </p>
                  <p
                    class="time"
                  >
                    07:27:57
                  </p>
                </section>,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="weekday"
                    />,
                    <p
                      className="date"
                    >
                      November 6, 2017
                    </p>,
                    <p
                      className="time"
                    >
                      07:27:57
                    </p>,
                  ],
                  "className": "goal-card-date-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <p
                      class="weekday"
                    />,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "weekday",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": <p
                      class="date"
                    >
                      November 6, 2017
                    </p>,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "November 6, 2017",
                      "className": "date",
                    },
                    "ref": null,
                    "rendered": Array [
                      "November 6, 2017",
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": <p
                      class="time"
                    >
                      07:27:57
                    </p>,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "07:27:57",
                      "className": "time",
                    },
                    "ref": null,
                    "rendered": Array [
                      "07:27:57",
                    ],
                    "type": "p",
                  },
                ],
                "type": "section",
              },
            ],
            "type": "section",
          },
          Object {
            "instance": <section
              class="card-right-side"
            >
              <p
                class="goal-description"
              >
                Go to gym
              </p>
              <button
                class="delete-button"
              >
                Delete
              </button>
            </section>,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="goal-description"
                >
                  Go to gym
                </p>,
                <button
                  className="delete-button"
                  onClick={[Function]}
                >
                  Delete
                </button>,
              ],
              "className": "card-right-side",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <p
                  class="goal-description"
                >
                  Go to gym
                </p>,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Go to gym",
                  "className": "goal-description",
                },
                "ref": null,
                "rendered": Array [
                  "Go to gym",
                ],
                "type": "p",
              },
              Object {
                "instance": <button
                  class="delete-button"
                >
                  Delete
                </button>,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete",
                  "className": "delete-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Delete",
                ],
                "type": "button",
              },
            ],
            "type": "section",
          },
        ],
        "type": "article",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
